mern-ecommerce/routes/product.js

```diff
@@ -18,6 +18,7 @@ const { requireSignin, isAuth, isAdmin } = require('../controllers/auth');
 const { userById } = require('../controllers/user');
 
 router.get('/product/:productId', read);
+router.put("/product/:productId", update);
```

Comment:
The `update` function is not defined in this file, it's imported from `../controllers/product`. Make sure it's properly defined and exported in the `product` controller.